.github/workflows/deploy.yml

```diff
@@ -7,46 +7,50 @@ on:
         - staging
 
 jobs:
-    deploy_instance_1:
+    deploy_instance_staging:
         if: github.ref == 'refs/heads/staging'
         runs-on: ubuntu-latest
 
         steps:
             - name: Checkout code
               uses: actions/checkout@v3
+            
+            - name: Setup SSH
+              run: |
+                mkdir -p ~/.ssh
+                echo "${{ secrets.SERVER_KEY }}" > ~/.ssh/id_ed25519 && chmod 600 ~/.ssh/id_ed25519
 
             - name: Deploy to Instance 1
-              uses: appleboy/ssh-action@v1.0.3
-              with:
-                host: ${{ secrets.STAGING_SERVER_HOST }}
-                username: ${{ secrets.STAGING_SERVER_USER }}
-                key: ${{ secrets.SERVER_KEY }}
-                script: |
-                    set -x
-                    cd /root/shutter-explorer
-                    git pull origin staging
-                    git submodule update --init --recursive
-                    cd docker
-                    docker compose up -d --build
-
-    deploy_instance_2:
+              run: >
```

Comment:
When you use the `>` block type in yaml all lines get concatenated. This won't work in this case because you have multiple commands.

So either change to `|` (but then the ssh command needs to be merged back into one line) or add `&&`s between the commands.